File: SchoolWebApp-Client/src/app/school/models/educationLevel.ts
import { Curriculum } from '@/academics/models/curriculum';
import {ResourceModel} from '@/core/models/ResourceModel';
import { EducationLevelType } from './education-level-types';

export class EducationLevel extends ResourceModel<EducationLevel> {
    public name?: string;
    public description?: string;
    public abbr?: string;
    public numOfYears?: number;

    public educationLevelTypeId?: number;
    public educationLevelType?: EducationLevelType;
    public curriculumId?: number;
    public curriculum?: Curriculum;

    constructor(model?: Partial<EducationLevel>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/School/EducationLevel.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.CBE.Assessments;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.School
{
    public class EducationLevel : Base
    {
        [Required(ErrorMessage = "Enter the education level name.")]
        [Display(Name = "Education level name")]
        [StringLength(255)]
        public required string Name { get; set; }
        [StringLength(255)]
        public string? Abbr { get; set; }
        public int NumOfYears { get; set; }

        public int Rank { get; set; }
        public string? Description { get; set; }

        public int EducationLevelTypeId { get; set; }
        public EducationLevelType? EducationLevelType { get; set; }
        public int CurriculumId { get; set; }
        public Curriculum? Curriculum { get; set; }

        public List<LearningLevel> LearningLevels { get; set; } = new();
        public List<EducationLevelSubject> educationLevelSubjects { get; set; } = new();
        public List<BroadOutcome> BroadOutcomes { get; set; } = new();
    }
}
